src/features/providers/lib/utils.ts

```diff
@@ -27,3 +28,23 @@ export function isProviderType(value: unknown): value is ProviderType {
 export function isProviderAuthType(value: unknown): value is ProviderAuthType {
   return Object.values(ProviderAuthType).includes(value as ProviderAuthType);
 }
+
+export function getProviderEndpointByAuthType(provider_type: ProviderType) {
+  return match(provider_type)
+    .with(ProviderType.OPENAI, () => "https://api.openai.com")
+    .with(ProviderType.ANTHROPIC, () => "https://api.anthropic.com")
+    .with(ProviderType.OPENROUTER, () => "https://openrouter.ai/api")
+    .with(ProviderType.OLLAMA, () => "http://host.docker.internal:11434")
+    .with(ProviderType.LM_STUDIO, () => "http://host.docker.internal:1234")
+    .with(ProviderType.LLAMACPP, () => "http://host.docker.internal:8080")
+    .with(ProviderType.VLLM, () => "http://host.docker.internal:8000")
+    .exhaustive();
+}
+
+export function getProviderAuthByType(provider_type: ProviderType) {
+  return match(provider_type)
+    .with(ProviderType.OPENAI, () => ProviderAuthType.API_KEY)
+    .with(ProviderType.ANTHROPIC, () => ProviderAuthType.API_KEY)
+    .with(ProviderType.OPENROUTER, () => ProviderAuthType.API_KEY)
+    .otherwise(() => ProviderAuthType.NONE);
```

Comment:
should this one not also be exhaustive instead? and the same point, not in the scope of this PR, but if there is hardcoded "data" or mapping should be done in BE, unless it's something that is 100% only relevant in the dashboard :thinking:

Reply:
the logic here is return api_key for those provider type, for the other return none, for that reason is correct to use otherwise rather than exhaustive. It will be none for the rest of the provider type, so duplicated a lot.

Also this is for prefilled fields in the form, so a UI logic, an helper, so the BE doesn't have this logic, they have only a validation in case of missing api_key for a provider